src/main/kotlin/net/lumalyte/lg/LumaGuilds.kt

```
@@ -979,6 +986,35 @@ class LumaGuilds : JavaPlugin() {
+            // provider exists, re-read channels.yml.
+            Bukkit.getScheduler().runTaskLater(this, Runnable {
+                try {
+                    channelManager.reload()
+                    logColored("✓ LumaGuilds RoseChat channels registered (guild, guild-ally, guild-modchat)")
+                } catch (e: Exception) {
+                    logger.warning("Failed to reload RoseChat channels after provider registration: ${e.message}")
+                }
+            }, 1L)
+        } catch (e: NoClassDefFoundError) {
+            logColored("⚠ RoseChat API classes unavailable — guild channels not registered")
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)